### ISSUE TITLE
Custom module map and module name yields wrong header locations for static Swift libs

### ISSUE THREAD

**Issue by zwaldowski** [CocoaPods/CocoaPods#7726]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

1. Create a pod that sets both `module_name` and `module_map`. (Aside: The `modulemap` is used to declare an explicit module for importing C. There are tons of other ways to accomplish this but none work quite as well. I welcome a recommendation.)
1. Point to it in a `Podfile`.
1. Set `Podfile` to _not_ have `use_frameworks!`.
1. Run `pod install`.
1. Open project workspace.
1. Add an `import` statement for the pod.
1. Build.

## What did you expect to happen?

The pod can compile by finding the umbrella header and embedded header like it can when `use_frameworks` is set.

## What happened instead?

<img width="228" alt="screen shot 2018-05-10 at 09 44 09" src="https://user-images.githubusercontent.com/170812/39881769-b97ad658-5436-11e8-92da-3f05d0bbc08b.png">

The headers are placed in a way the module map can't find.

## CocoaPods Environment

<details>

### Stack

```
   CocoaPods : 1.5.2
        Ruby : ruby 2.5.0p0 (2017-12-25 revision 61468) [x86_64-darwin17]
    RubyGems : 2.7.6
        Host : Mac OS X 10.13.4 (17E202)
       Xcode : 9.4 (9Q1019a)
         Git : git version 2.17.0
Ruby lib dir : /usr/local/var/rbenv/versions/2.5.0/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 52d858d1c0afcdca860a9bc74c61ba037f5a61aa
```

### Installation Source

```
Executable Path: /usr/local/var/rbenv/versions/2.5.0/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
platform :ios, '11.0'

use_modular_headers!

target 'PodIntegration' do
  pod 'PodKit', :path => './PodKit'
end
```
</details>

## Project that demonstrates the issue

https://bucket.waldowski.me/PodIntegration-TeLTCPNizH

The app target is `PodIntegration.` The pod is named `PodKit`, for a library named `Kit` which requires a private-ish C module `CKit`.

**Comment by segiddins** [comment]
Yup, known issue, don't know how to fix it besides copying all headers in the build phase

**Comment by zwaldowski** [comment]
Do note that it fails to compile while building the static lib - not even getting to the part where the Copy Headers Build Phase comes into play.

**Comment by ileitch** [comment]
I'm running into this issue too, with https://github.com/uber/ios-snapshot-test-case and https://github.com/Iterable/swift-sdk. Neither define a custom `module_map`.
